app/controllers/fhir/r4/api_controller.rb

```diff
@@ -0,0 +1,374 @@
+            ],
+            searchParam: [
+              FHIR::CapabilityStatement::Rest::Resource::SearchParam.new(name: 'family', type: 'string'),
+              FHIR::CapabilityStatement::Rest::Resource::SearchParam.new(name: 'given', type: 'string'),
+              FHIR::CapabilityStatement::Rest::Resource::SearchParam.new(name: 'telecom', type: 'string'),
+              FHIR::CapabilityStatement::Rest::Resource::SearchParam.new(name: 'email', type: 'string')
+            ]
+          ),
+          FHIR::CapabilityStatement::Rest::Resource.new(
+            type: 'Observation',
+            interaction: [
+              FHIR::CapabilityStatement::Rest::Resource::Interaction.new(code: 'read')
+            ]
+          ),
+          FHIR::CapabilityStatement::Rest::Resource.new(
+            type: 'QuestionnaireResponse',
+            interaction: [
+              FHIR::CapabilityStatement::Rest::Resource::Interaction.new(code: 'read')
+            ]
+          )
+        ]
+      )
+    )
+    status_ok(resource) && return
+  rescue StandardError
+    render json: operation_outcome_fatal.to_json, status: :internal_server_error
+  end
+
+  # Handle OPTIONS requests for CORS preflight
+  def options
+    render plain: ''
+  end
+
+  private
+
+  # Current user account as authenticated via doorkeeper
+  def current_resource_owner
+    User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+  end
+
+  # Current user account can use api
+  def check_api_access
+    current_resource_owner&.can_use_api?
+  end
+
+  # Generic 400 bad request response
+  def status_bad_request
+    respond_to do |format|
+      format.any { head :bad_request }
+    end
+  end
+
+  # Generic 403 forbidden response
+  def status_forbidden
+    respond_to do |format|
+      format.any { head :forbidden }
+    end
+  end
+
+  # Generic 201 created response
+  def status_created(resource)
+    respond_to do |format|
+      format.json { render json: resource.to_json, status: :created, location: full_url_helper(resource), content_type: 'applicaton/fhir+json' }
+      format.fhir_json { render json: resource.to_json, status: :ok, location: full_url_helper(resource), content_type: 'applicaton/fhir+json' }
+    end
+  end
+
+  # Generic 200 okay response
+  def status_ok(resource)
+    respond_to do |format|
+      format.json { render json: resource.to_json, status: :ok, content_type: 'applicaton/fhir+json' }
+      format.fhir_json { render json: resource.to_json, status: :ok, content_type: 'applicaton/fhir+json' }
+    end
+  end
+
+  # Get a patient by id
+  def get_patient(id)
+    current_resource_owner.viewable_patients.find_by(id: id)
+  end
+
+  # Search for patients
+  def search_patients(options)
+    query = current_resource_owner.viewable_patients
+    options.each do |option, search|
+      case option
+      when 'family'
+        query = query.where('last_name like ?', "#{search}%")
```

Comment:
Is this the expected search algorithm for searching for patients on FHIR servers? Seems a little weird that you could do things like get patients whose first name starts with a given letter, or phone number that starts with some number.

Reply:
https://www.hl7.org/fhir/search.html#string, it's also how we do search in the UI.